frontend/apps/monkvision/components/chart-box/chart-box.mjs

```diff
@@ -223,5 +227,19 @@ async function _getContent(api, params) {
 
 const _isTrue = string => string?string.toLowerCase() == "true":false;
 
-export const chart_box = {trueWebComponentMode: true, elementRendered, setTimeRange, getTimeRange}
+async function exportCSV(elementId) {
+	const element = document.querySelector("#maincontent > page-generator").shadowRoot.querySelector(`#${elementId}`);
+	const csvParams = element.getAttribute("params");
+	const content =  await _getContent("exportCSV", csvParams);
+
+	let hiddenElement = document.createElement('a');
+	hiddenElement.href = 'data:text/csv;charset=utf-8,' + encodeURI(content.contents.csvData);
+	hiddenElement.target = '_blank';
+
+	//provide the name for the CSV file to be downloaded
+	hiddenElement.download = `${element.getAttribute("elementTitle")}-${dateAsHTMLDateValue(new Date)}.csv`;
```

Comment:
The page name should be added to the CSV file along with the selected time range to make it easier to recognise where the data originated from.